lib/ansible/modules/cloud/docker/docker_node_facts.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/python
+#
+# (c) 2019 Piotr Wojciechowski <piotr@it-playground.pl>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: docker_node_facts
+
+short_description: Retrieves facts about docker swarm node from Swarm Manager
+
+description:
+  - Retrieves facts about a docker node.
+  - Essentially returns the output of C(docker node inspect <name>)
+  - Must be executed on host running as Swarm Manager
+
+version_added: "2.8"
+
+options:
+  name:
+    description:
+      - The name of the node to inspect.
+      - When identifying an existing node name may be a name or a long or short container ID.
+    required: true
+extends_documentation_fragment:
+    - docker
+
+author:
+    - Piotr Wojciechowski (@wojciechowskipiotr)
+
+requirements:
+    - python >= 2.7
+    - "docker >= 2.6.0"
```

Comment:
In fact, this has been supported since docker-py 1.10.0. There's no need to require docker-py 2.6.0.

(https://github.com/docker/docker-py/commit/a75553b3)